src/Button/variants.js

```diff
@@ -1,67 +1,166 @@
+  justifyContent: 'center',
+  width: 'fit-content',
+  label: {
+    cursor: 'pointer',
+    pointerEvents: 'none'
+  },
+  ':hover': {
+    opacity: 0.75
+  },
+  ':active ': {
+    transform: 'scale(0.965)'
+  },
+  ':focus': {
+    opacity: 0.75,
+    outlineStyle: 'none',
+    outlineColor: 'transparent'
+  },
+  ':disabled': {
+    opacity: 0.3,
+    transform: 'scale(1)',
+    cursor: 'not-allowed'
+  }
+}
+const PRIMARY_DEFAULT = {
+  ...BUTTON_DEFAULT,
+  backgroundColor: colors.brand,
+  color: colors.white,
+  ':hover': {
+    backgroundColor: colors.brandDark
+  },
+  ':active': {
+    backgroundColor: colors.brandDarkest
+  },
+  ':focus': {
+    opacity: 1
+  },
+  ':disabled': {
+    backgroundColor: colors.disabled,
+    transform: 'scale(1)',
+    cursor: 'not-allowed'
+  }
+}
+const PRIMARY_STYLE = {
+  ...PRIMARY_DEFAULT
+}
+const LARGE_STYLE = {
+  ...PRIMARY_DEFAULT,
+  minWidth: sizes.largeButtonMinWidth
+}
+
+const OUTLINE_DEFAULT = {
+  ...BUTTON_DEFAULT,
+  color: colors.text,
+  backgroundColor: 'transparent',
+  border: `${BORDERS[1]}`
+}
+const OUTLINE_STYLE = {
+  ...OUTLINE_DEFAULT,
+  borderColor: colors.accent
+}
+
+const POSITIVE_STYLE = {
+  ...BUTTON_DEFAULT,
+  backgroundColor: colors.positive,
+  color: 'white'
+}
+const LINK_DEFAULT = {
+  ...BUTTON_DEFAULT,
   background: 'none',
   minWidth: 'max-content',
   height: 'auto',
   padding: 0,
-  minHeight: 'max-content'
-})
-
-export default {
-  primary: {
-    backgroundColor: 'accent',
-    color: 'white'
-  },
-  outline: {
-    color: 'text',
-    backgroundColor: 'transparent',
-    borderWidth: '1px',
-    borderStyle: 'solid',
-    borderColor: 'accent'
+  minHeight: 'max-content',
+  ':hover': {
+    color: colors.brandDark
   },
-  outlineAlt: {
-    color: 'text',
-    backgroundColor: 'transparent',
-    borderWidth: '1px',
-    borderStyle: 'solid',
-    borderColor: 'text'
+  ':active': {
+    color: colors.brandDarkest
   },
-  positive: {
-    backgroundColor: 'positive',
-    color: 'white'
+  ':disabled': {
+    color: colors.disabled,
+    transform: 'scale(1)',
+    cursor: 'not-allowed'
+  }
+}
+const LINK_STYLE = {
+  ...LINK_DEFAULT,
+  color: colors.brand
+}
+const LINK_POSITIVE_STYLE = {
+  ...LINK_DEFAULT,
+  color: colors.positive,
+  ':hover': {
+    opacity: 0.75
   },
-  link: linkStyles('accent'),
-  linkSecondary: linkStyles('lighterText'),
-  linkPositive: linkStyles('positive')
+  ':active': {
+    color: colors.positive
+  }
+}
+export default {
+  primary: PRIMARY_STYLE,
+  large: LARGE_STYLE,
+  link: LINK_STYLE,
+  outline: OUTLINE_STYLE,
+  outlineAlt: OUTLINE_STYLE,
+  positive: POSITIVE_STYLE,
+  linkSecondary: LINK_STYLE,
+  linkPositive: LINK_POSITIVE_STYLE
 }
 
-export const spinnerVariants = {
+export const iconVariants = {
   primary: {
-    background: 'accent',
-    primary: 'white'
-  },
-  outline: {
-    background: 'white',
-    primary: 'accent'
+    color: colors.white,
+    fontSize: '1'
```

Comment:
I think this is an index, we should use integer? Please check this:
```suggestion
    fontSize: 1
```